planning/behavior_path_planner/src/scene_module/lane_change/util.cpp

```diff
@@ -739,4 +724,172 @@ std::vector<DrivableLanes> generateDrivableLanes(
   return drivable_lanes;
 }
 
+std::optional<LaneChangePath> getAbortPaths(
+  const std::shared_ptr<const PlannerData> & planner_data, const LaneChangePath & selected_path,
+  [[maybe_unused]] const Pose & ego_pose_before_collision,
+  const BehaviorPathPlannerParameters & common_param,
+  [[maybe_unused]] const LaneChangeParameters & lane_change_param)
+{
+  const auto & route_handler = planner_data->route_handler;
+  const auto current_speed = util::l2Norm(planner_data->self_odometry->twist.twist.linear);
+  const auto current_pose = planner_data->self_pose->pose;
+  const auto current_lanes = selected_path.reference_lanelets;
+
+  const auto ego_nearest_dist_threshold = planner_data->parameters.ego_nearest_dist_threshold;
+  const auto ego_nearest_yaw_threshold = planner_data->parameters.ego_nearest_yaw_threshold;
+
+  auto resampled_selected_path = selected_path.path;
+
+  const auto ego_pose_idx = motion_utils::findFirstNearestIndexWithSoftConstraints(
+    resampled_selected_path.points, current_pose, ego_nearest_dist_threshold,
+    ego_nearest_yaw_threshold);
+  const auto lane_changing_end_pose_idx = motion_utils::findFirstNearestIndexWithSoftConstraints(
+    resampled_selected_path.points, resampled_selected_path.points.back().point.pose,
+    ego_nearest_dist_threshold, ego_nearest_yaw_threshold);
+
+  const auto pose_idx_min = [&](const double param_time, double & turning_point_dist) {
+    if (ego_pose_idx > lane_changing_end_pose_idx) {
+      return ego_pose_idx;
+    }
+
+    const auto desired_distance = std::max(2.77, current_speed) * param_time;
+    const auto & points = resampled_selected_path.points;
+    size_t idx{0};
+    for (idx = ego_pose_idx; idx < lane_changing_end_pose_idx; ++idx) {
+      const auto dist_to_ego =
+        util::getSignedDistance(current_pose, points.at(idx).point.pose, current_lanes);
+      turning_point_dist = dist_to_ego;
+      if (dist_to_ego > desired_distance) {
+        break;
+      }
+    }
+    return idx;
+  };
+
+  const auto abort_delta_time = lane_change_param.abort_delta_time;
+
+  double abort_start_dist{0.0};
+  const auto abort_start_idx = pose_idx_min(abort_delta_time, abort_start_dist);
```

Comment:
how about make the output of `pose_idx_min` std::pair and return both `abort_start_dist` and `abort_start_idx`?

```
const auto pose_idx_min = [&](const double param_time)
{
   ...
   return std::pair{idx, turning_point_dist};
};
```

Then you can receive two outputs like this:
```
const auto [abort_start_dist, abort_start_idx] = pose_idx_min(abort_delta_time);
```

Reply:
this was fixed in [4b61a8f](https://github.com/autowarefoundation/autoware.universe/pull/2359/commits/4b61a8f806b49cc4881b48255cbc89e6be65ac66)